p2p/server/server.go

```diff
@@ -172,38 +172,42 @@ func (s *Server) Run(ctx context.Context) error {
 				return nil
 			}
 			eg.Go(func() error {
-				s.queueHandler(ctx, req.stream)
+				ok := s.queueHandler(ctx, req.stream)
 				if s.metrics != nil {
 					s.metrics.serverLatency.Observe(time.Since(req.received).Seconds())
-					s.metrics.completed.Inc()
+					if ok {
+						s.metrics.completed.Inc()
+					} else {
+						s.metrics.failed.Inc()
+					}
 				}
 				return nil
 			})
 		}
 	}
 }
 
-func (s *Server) queueHandler(ctx context.Context, stream network.Stream) {
+func (s *Server) queueHandler(ctx context.Context, stream network.Stream) bool {
```

Comment:
How about returning the error instead? I find returning bools confusing as it's difficult to immediately say what they mean.

Reply:
I'm unsure about it. Currently, `queueHandler` logs some of the errors with additional details:
```
		s.logger.With().Warning("request limit overflow",
			log.Int("limit", s.requestLimit),
			log.Uint64("request", size),
```
while others are silently skipped b/c they can be just too frequent (perhaps we could add `debug` level logging for them). So if I return the error from `queueHandler` instead of the bool we either
* ignore the error itself and just check that it's not `nil` in `Run()`, which basically makes returning the error meaningless
* or we log the error in `Run` and add logic to check error type via `errors.Is` there to see if/how the error needs to be logged, but in addition to some complication of the logic, this also removes the advantage of structured logging, b/c e.g. in the case above `limit` and more importantly `request` will be mere part of the error message and not separate log fields